x-pack/packages/kbn-elastic-assistant/impl/assistant/quick_prompts/quick_prompt_settings_management/index.tsx

```diff
@@ -31,31 +32,65 @@ import {
 } from '../../common/components/assistant_settings_management/pagination/use_session_pagination';
 import { QUICK_PROMPT_TABLE_SESSION_STORAGE_KEY } from '../../../assistant_context/constants';
 import { useAssistantContext } from '../../../assistant_context';
+import {
+  DEFAULT_CONVERSATIONS,
+  useSettingsUpdater,
+} from '../../settings/use_settings_updater/use_settings_updater';
+import { useFetchPrompts } from '../../api';
+
+const QuickPromptSettingsManagementComponent = () => {
+  const { nameSpace, basePromptContexts, toasts } = useAssistantContext();
+
+  const { data: allPrompts } = useFetchPrompts();
+
+  const {
+    promptsBulkActions,
+    quickPromptSettings,
+    resetSettings,
+    saveSettings,
+    setPromptsBulkActions,
+    setUpdatedQuickPromptSettings,
+  } = useSettingsUpdater(
+    DEFAULT_CONVERSATIONS, // Quick Prompt settings do not require conversations
+    allPrompts,
+    false // Quick Prompt settings do not require conversations
+  );
+
+  // Quick Prompt Selection State
+  const [selectedQuickPrompt, setSelectedQuickPrompt] = useState<PromptResponse | undefined>();
+  const onSelectedQuickPromptChange = useCallback((quickPrompt?: PromptResponse) => {
+    setSelectedQuickPrompt(quickPrompt);
+  }, []);
 
-interface Props {
-  handleSave: (shouldRefetchConversation?: boolean) => void;
-  onCancelClick: () => void;
-  onSelectedQuickPromptChange: (quickPrompt?: PromptResponse) => void;
-  quickPromptSettings: PromptResponse[];
-  resetSettings?: () => void;
-  selectedQuickPrompt: PromptResponse | undefined;
-  setUpdatedQuickPromptSettings: React.Dispatch<React.SetStateAction<PromptResponse[]>>;
-  promptsBulkActions: PromptsPerformBulkActionRequestBody;
-  setPromptsBulkActions: React.Dispatch<React.SetStateAction<PromptsPerformBulkActionRequestBody>>;
-}
-
-const QuickPromptSettingsManagementComponent = ({
-  handleSave,
-  onCancelClick,
-  onSelectedQuickPromptChange,
-  quickPromptSettings,
-  resetSettings,
-  selectedQuickPrompt,
-  setUpdatedQuickPromptSettings,
-  promptsBulkActions,
-  setPromptsBulkActions,
-}: Props) => {
-  const { nameSpace, basePromptContexts } = useAssistantContext();
+  useEffect(() => {
+    if (selectedQuickPrompt != null) {
+      setSelectedQuickPrompt(quickPromptSettings.find((q) => q.name === selectedQuickPrompt.name));
+    }
+  }, [quickPromptSettings, selectedQuickPrompt]);
+
+  const quickPrompts = useMemo(
+    () =>
+      quickPromptSettings.length === 0
+        ? allPrompts.data.filter((p) => p.promptType === PromptTypeEnum.quick)
+        : quickPromptSettings,
+    [allPrompts.data, quickPromptSettings]
+  );
+
+  const handleSave = useCallback(
+    async (param?: { callback?: () => void }) => {
+      await saveSettings();
+      toasts?.addSuccess({
+        iconType: 'check',
+        title: SETTINGS_UPDATED_TOAST_TITLE,
+      });
+      param?.callback?.();
+    },
```

Comment:
Looks like there's a bug here with handling failed requests. The request here `409`'d because it already exists, but it seems like it's successful and shows up in the table. There's eventual consistency and it'll disappear after trying to edit again though.

Also, that `409` seems suspect as it shows two `id`'s in the message:

```
prompt with id: \"RUJXmZAB29PPJeuQ9hwt,RkJXmZAB29PPJeuQ9hwt\" already exists"
```

And the bulk update payload seems suspect too, as I was only updating a single prompt, but it included four different ones that I had previously created:

<img width="949" alt="image" src="https://github.com/elastic/kibana/assets/2946766/52554c6e-adab-43d2-b98c-2ea27fcbf691">

Reply:
Yeah there's some stale state here or something. Any subsequent create fails with that same dupe id error. Once I deleted all the custom quick prompts it seems to be working again.